com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs

```diff
@@ -552,32 +555,58 @@ void SortCameras(Camera[] cameras)
             return desc;
         }
 
-        static Lightmapping.RequestLightsDelegate lightsDelegate = (Light[] requests, NativeArray<LightDataGI> lightsOutput) =>
+        private static Lightmapping.RequestLightsDelegate lightsDelegate = (Light[] requests, NativeArray<LightDataGI> lightsOutput) =>
         {
             LightDataGI lightData = new LightDataGI();
 #if UNITY_EDITOR
             // Always extract lights in the Editor.
             for (int i = 0; i < requests.Length; i++)
             {
                 Light light = requests[i];
+
+                var additionalLightData = light.GetComponent<UniversalAdditionalLightData>();
+                if (additionalLightData == null)
+                {
+                    additionalLightData = ComponentSingleton<UniversalAdditionalLightData>.instance;
+                }
+
+                Cookie cookie;
+                LightmapperUtils.Extract(light, out cookie);
+
                 switch (light.type)
                 {
                     case LightType.Directional:
                         DirectionalLight directionalLight = new DirectionalLight();
                         LightmapperUtils.Extract(light, ref directionalLight);
-                        lightData.Init(ref directionalLight);
+
+                        if (light.cookie != null && additionalLightData != null)
+                        {
+                            // Size == 1 / Scale
+                            cookie.sizes = additionalLightData.lightCookieSize;
+                            // Offset, Map cookie UV offset to light position on along local axes.
+                            if (additionalLightData.lightCookieOffset != Vector2.zero)
+                            {
+                                var r = light.transform.right * additionalLightData.lightCookieOffset.x;
+                                var u = light.transform.up * additionalLightData.lightCookieOffset.y;
```

Comment:
The light scale seems to be ignored and has now effect based on my testing.
I set it in a way that UV offset is directly mapped moving the light on it's local axes, since the GI doesn't support UV offset.

Reply:
Good, I just wanted to check we are conscious of the behavior we are choosing